drivers: memc: memc_nxp_flexram.h: Move to the public includes directory

jacob-wienecke-nxp opened the pull request:
Moved to: include/zephyr/drivers/misc/flexram/memc_nxp_flexram.h

This change makes it so that the .h file does not need to be pulled in using the CMakeLists.txt file and can be included like other public includes.

Removes drivers/memc/memc_nxp_flexram.h

Add memc_nxp_flexram.h to include/zephyr/drivers/misc/flexram

Modify drivers/memc/memc_nxp_flexram.c to use the new include path.

Modify the mimxrt1170 magic_addr sample to include the driver using the new include path.

Modify the soc file: soc/nxp/imxrt/imxrt11xx/soc.c to use the new path.

Add relevant information to migration-guide-4.1.rst.

jacob-wienecke-nxp committed:
drivers: memc: memc_nxp_flexram.h: Move to the public includes directory

Moved to: include/zephyr/drivers/misc/flexram/memc_nxp_flexram.h

This change makes it so that the .h file does not need to be pulled in
using the CMakeLists.txt file, and can be included like other public
includes.

Removes drivers/memc/memc_nxp_flexram.h

Add memc_nxp_flexram.h to include/zephyr/drivers/misc/flexram

Modify drivers/memc/memc_nxp_flexram.c to use the new include path.

Modifies the mimxrt1170 magic_addr sample to include the driver using
the new include path.

Modify the soc file: soc/nxp/imxrt/imxrt11xx/soc.c to use the new path.

Add relevant information to migration-guide-4.2.rst.

Signed-off-by: Jacob Wienecke <jacob.wienecke@nxp.com>
Co-authored-by: Declan Snyder <declan.snyder@nxp.com>

kartben reviewed (changes requested):
Needs to update 4.2 migration guide instead. Thanks!

Commit:
drivers: Move flexram to misc driver

Flexram is really not a memory controller, and does not belong in memc
namespace or directory. Move it to it's own misc directory and remove
memc_ from the namespace.

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

decsny commented:
update: 
- rebase
- fix migration guide
- fix bug where rt10xx soc.c did not have the update to the header
- change include file name to "nxp_flexram.h" instead of "memc_nxp_flexram.h" since it was being put in the misc directory, the namespace of memc didn't make sense
- move the .c file also to a misc driver directory and move all flexram things out of memc
- remove memc namespace from all API and kconfig

dleach02 commented on doc/releases/migration-guide-4.2.rst:
```diff
@@ -202,6 +202,17 @@ Stepper
+Misc
+====
+
+* Moved file `drivers/memc/memc_nxp_flexram.h` to
```
@jacob-wienecke-nxp I'm guessing this check error wants you to use double backticks (@kartben correct?)

kartben replied:
Yep

jacob-wienecke-nxp replied:
Addressed yesterday doing a rebase and amending the commit.

This effectively changes line 210 to the requested change. Unless I missed something else?

jacob-wienecke-nxp replied:
I misunderstood

It seems like line 208 is also requiring that change.

Apologies for the delay.
